mailpoet/lib/Automation/Engine/Control/StepRunner.php

```diff
@@ -116,6 +116,11 @@ private function runStep(array $args): void {
       return;
     }
 
+    // no need to schedule a new step if the next step is null
+    if (empty($nextStepArgs[0]['step_id'])) {
```

Comment:
I think you need to set the status of the workflowrun to `WorkflowRun::STATUS_COMPLETE` which currently happens in L91 when no stepId was found.

Reply:
Good catch! I added completion of the workflow run here.